### PR TITLE
Add GrindstoneInventoryMock and Tests for it

### PR DESCRIPTION
**Pull request by thelooter**
# Description
Implement the GrindstoneInventory and add Tests for it

# Checklist
The following items should be checked before the pull request can be merged.
- [x] Unit tests added.
- [x] Code follows existing code format.

# Info on creating a pull request
- Make sure that unit tests are added which test the relevant changes.
- Make sure that the changes follow the existing code format.

# For maintainer
When a PR is approved, the maintainer should label this PR with `release/*`.

- If the PR fixes a bug in MockBukkit, update the patch version. (if the current version is `0.4.1`, the new version should be `0.4.2`)
- If the PR adds a new feature to MockBukkit, update minor version. (if the current version is `0.4.1`, the new version should be `0.5.0`)

Note that a PR that fixes an `UnimplementedOperationException` should be considered a new version and not a bugfix.

### CODE REVIEW

File: src/main/java/be/seeseemelk/mockbukkit/inventory/GrindstoneInventoryMock.java [hunk 1]
```diff
@@ -0,0 +1,21 @@
+package be.seeseemelk.mockbukkit.inventory;
+
+import org.bukkit.event.inventory.InventoryType;
+import org.bukkit.inventory.InventoryHolder;
+import org.jetbrains.annotations.NotNull;
+
+public class GrindstoneInventoryMock extends InventoryMock
+{
+	public GrindstoneInventoryMock(InventoryHolder holder)
```
Comment (Insprill) [inline src/main/java/be/seeseemelk/mockbukkit/inventory/GrindstoneInventoryMock.java]:
```suggestion
{

	public GrindstoneInventoryMock(InventoryHolder holder)
```

File: src/main/java/be/seeseemelk/mockbukkit/inventory/GrindstoneInventoryMock.java [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+	}
+
+	@Override
+	public @NotNull InventoryMock getSnapshot()
+	{
+		GrindstoneInventoryMock inventory = new GrindstoneInventoryMock(getHolder());
+		inventory.setContents(getContents());
+		return inventory;
+	}
+}
```
Comment (Insprill) [inline src/main/java/be/seeseemelk/mockbukkit/inventory/GrindstoneInventoryMock.java]:
```suggestion
	}

}
```

File: src/test/java/be/seeseemelk/mockbukkit/inventory/GrindstoneInventoryMockTest.java [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+	@Test
+	public void testGetSnapshot()
+	{
+		InventoryMock snapshot = inventory.getSnapshot();
+
+		assertThat(snapshot, instanceOf(GrindstoneInventoryMock.class));
+		assertThat(snapshot, not(equalToObject(inventory)));
+	}
+}
```
Comment (Insprill) [inline src/test/java/be/seeseemelk/mockbukkit/inventory/GrindstoneInventoryMockTest.java]:
```suggestion
	}

}
```

File: src/test/java/be/seeseemelk/mockbukkit/inventory/GrindstoneInventoryMockTest.java [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+import org.junit.jupiter.api.Test;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalToObject;
+import static org.hamcrest.Matchers.instanceOf;
+import static org.hamcrest.Matchers.not;
+
+public class GrindstoneInventoryMockTest
+{
+	private GrindstoneInventoryMock inventory;
```
Comment (Insprill) [inline src/test/java/be/seeseemelk/mockbukkit/inventory/GrindstoneInventoryMockTest.java]:
```suggestion
{

	private GrindstoneInventoryMock inventory;
```

**Comment by thelooter** [comment]
Also adresses Issue #113

File: src/main/java/be/seeseemelk/mockbukkit/inventory/GrindstoneInventoryMock.java [hunk 5]
```diff
@@ -0,0 +1,21 @@
+package be.seeseemelk.mockbukkit.inventory;
+
+import org.bukkit.event.inventory.InventoryType;
+import org.bukkit.inventory.InventoryHolder;
+import org.jetbrains.annotations.NotNull;
+
+public class GrindstoneInventoryMock extends InventoryMock
```
Comment (Insprill) [inline src/main/java/be/seeseemelk/mockbukkit/inventory/GrindstoneInventoryMock.java:7]: You should create another PR implementing a `ResultInventoryMock` we can extend from so it's code doesn't need to be duplicated across multiple inventory mocks. [CraftBukkit source](https://hub.spigotmc.org/stash/projects/SPIGOT/repos/craftbukkit/browse/src/main/java/org/bukkit/craftbukkit/inventory/CraftResultInventory.java).

Comment (Insprill) [inline src/main/java/be/seeseemelk/mockbukkit/inventory/GrindstoneInventoryMock.java:7]:
```suggestion
public class GrindstoneInventoryMock extends InventoryMock implements GrindstoneInventory
```

File: src/test/java/be/seeseemelk/mockbukkit/inventory/GrindstoneInventoryMockTest.java [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+		MockBukkit.unmock();
+	}
+
+	@Test
+	public void testGetSnapshot()
+	{
+		InventoryMock snapshot = inventory.getSnapshot();
+
+		assertThat(snapshot, instanceOf(GrindstoneInventoryMock.class));
+		assertThat(snapshot, not(equalToObject(inventory)));
```
Comment (Insprill) [inline src/test/java/be/seeseemelk/mockbukkit/inventory/GrindstoneInventoryMockTest.java]:
```suggestion
		assertInstanceOf(GrindstoneInventoryMock.class, snapshot);
		assertNotEquals(inventory, snapshot));
```

**Review by Insprill — CHANGES_REQUESTED** [review]

File: src/main/java/be/seeseemelk/mockbukkit/inventory/GrindstoneInventoryMock.java (hunk 5 above)
Comment (thelooter) [reply src/main/java/be/seeseemelk/mockbukkit/inventory/GrindstoneInventoryMock.java:7]: After consideration, we decided not to pursue this

File: src/main/java/be/seeseemelk/mockbukkit/inventory/GrindstoneInventoryMock.java [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+	}
+
+	@Override
+	public @NotNull InventoryMock getSnapshot()
+	{
+		GrindstoneInventoryMock inventory = new GrindstoneInventoryMock(getHolder());
+		inventory.setContents(getContents());
+		return inventory;
+	}
+}
```
Comment (Insprill) [inline src/main/java/be/seeseemelk/mockbukkit/inventory/GrindstoneInventoryMock.java]:
```suggestion
	}

}
```

**Review by Insprill — CHANGES_REQUESTED** [review]

**Review by Insprill — APPROVED** [review]

**Pull request merged**